Neo4j.Driver/Neo4j.Driver.Tests/Internal/IO/ValueSerializers/UnsupportedTypeSerializerTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) "Neo4j"
+// Neo4j Sweden AB [https://neo4j.com]
+// 
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using FluentAssertions;
+using Neo4j.Driver.Internal.IO;
+using Neo4j.Driver.Internal.IO.ValueSerializers;
+using Neo4j.Driver.Internal.Protocol;
+using Xunit;
+
+namespace Neo4j.Driver.Tests.Internal.IO.ValueSerializers;
+
+public class UnsupportedTypeSerializerTests : PackStreamSerializerTests
+{
+    internal override IPackStreamSerializer SerializerUnderTest { get; } = new UnsupportedTypeSerializer();
+
+    [Fact]
+    public void ShouldDeserialize()
+    {
+        var writerMachine = CreateWriterMachine();
+        var writer = writerMachine.Writer;
+
+        writer.WriteStructHeader(4, (byte)'?');
```

Comment:
Might be worth commenting on what each part is, or at least where to find the meaning of the magic numbers?